libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -217,6 +217,10 @@ impl BTCReceiveSwap {
             return Err(SwapError::Generic(anyhow!("Wrong address: {address_str}")));
         }
 
+        let max_allowed_deposit = std::cmp::min(
```

Comment:
Now that we calculate the max_allowed_deposit I think we should check if it is bellow min_allowed_deposit and return error in such case.

Reply:
Done